lib/eco/test/test_eco.py

```diff
@@ -2283,6 +2283,137 @@ def _update_from_pos_one_colour():
         self.treemanager.key_ctrl_z()
         self.treemanager.key_ctrl_z()
 
+    def test_undo_random_deletion_bug8(self):
+        # This is a test for a bug in the out-of-context analysis, where the
+        # boundaries of the subtree being parsed where not clear so the parser
+        # would reduce and reuse nodes outside of the original subtree. After
+        # the analysis failed, those reused subtrees would then not be reverted
+        # causing errors with detached subtrees.
```

Comment:
Should "with detached subtrees" be something like "because subtrees had become detached from the main parse tree"?